### PR TITLE
refact(volume-mgmt): replace replica status with mode

### PR DESCRIPTION
**Pull request by prateekpandey14**
the response status parameter for cmd `istgtcontrol -q replica` has been changed from status-->mode, this commit will replace the status parameter with mode in ReplicaStatus struct

Signed-off-by: prateekpandey14 <prateekpandey14@gmail.com>

### CODE REVIEW

**Comment by prateekpandey14** [comment]
@vishnuitta added 1 more positive test case with latest response data.

File: cmd/cstor-volume-mgmt/volume/volume_test.go [hunk 1]
```diff
@@ -284,6 +284,43 @@ func TestExtractReplicaStatusFromJSON(t *testing.T) {
 			},
 			true,
 		},
+		"valid single replica healthy status": {
```
Comment (vishnuitta) [inline cmd/cstor-volume-mgmt/volume/volume_test.go]: Use this sample output:
```{
   "volumeStatus" : [
      {
         "name" : "vol1",
         "status" : "Healthy",
         "replicaStatus" : [
            {
               "replicaId" : "14367084317571389083",
               "inflightSync" : "0",
               "inflightWrite" : "0",
               "Address" : "192.168.1.23",
               "upTime" : 5968,
               "checkpointedIOSeq" : "189",
               "inflightRead" : "0",
               "Mode" : "Healthy"
            }
         ]
      }
   ]
}
```

File: pkg/apis/openebs.io/v1alpha1/cstor_volume.go [hunk 2]
```diff
@@ -82,7 +82,7 @@ type CVStatus struct {
 // ReplicaStatus stores the status of replicas
 type ReplicaStatus struct {
 	ID                string `json:"replicaId"`
-	Status            string `json:"status"`
+	Mode              string `json:"mode"`
```
Comment (AmitKumarDas) [inline pkg/apis/openebs.io/v1alpha1/cstor_volume.go]: This seems to be a breaking change?
Can we have both the fields for backward compatibility?

Comment (vishnuitta) [reply pkg/apis/openebs.io/v1alpha1/cstor_volume.go]: end users won't be using this. So, should be fine.
But, need to find a way of not to have frequent changes in this.

Comment (prateekpandey14) [reply pkg/apis/openebs.io/v1alpha1/cstor_volume.go]: yes we are not using replica-status any where , its just a part of the CVR object.

**Pull request merged**